### PR TITLE
F/129835 published s123 survey support v2

### PR DESCRIPTION
**Pull request by rweber-esri**
Resubmitting this PR based off @dbouwman [branch](f/streamline-post-process) that refactors some of the post processing logic.

Survey123 `Feature Service` items have a `typeKeyword` that references the related `Form` item ID. Since, the `Feature Service` items are dependencies of the `Form` item, we need to post process them after the `Form` item is created.

### CODE REVIEW

**Comment by rweber-esri** [comment]
** CURRENTLY POINTING TO @dbouwman [branch/PR](https://github.com/Esri/solution.js/pull/343). Once that's merged, can update this to point to `develop`.

File: packages/feature-layer/src/feature-layer.ts [hunk 1]
```diff
@@ -328,6 +329,30 @@ export function createItemFromTemplate(
...
 }
 
+export function postProcess(
+  itemId: string,
+  type: string,
+  templates: common.IItemTemplate[],
+  templateDictionary: any,
+  authentication: common.UserSession
+): Promise<any> {
+  return Promise.all([
```
Comment (rweber-esri) [inline packages/feature-layer/src/feature-layer.ts:339]: @dbouwman per your request, using lower-level utils to fetch the item & data vs `deployUtils._getSolutionTemplateItem`

Comment (dbouwman) [reply packages/feature-layer/src/feature-layer.ts:339]: 👍

File: packages/common/src/templatization.ts [hunk 2]
```diff
@@ -319,7 +320,7 @@ export function templatizeIds(obj: any): any {
   const idTest: RegExp = /[0-9A-F]{32}/gim;
   if (obj && idTest.test(objString)) {
     // Templatize ids
-    const ids: string[] = objString.match(idTest) as string[];
+    const ids: string[] = objString.match(idTest).filter(unique);
```
Comment (rweber-esri) [inline packages/common/src/templatization.ts:323]: This fixes:

```
const results = templatizeIds({
  a: "ab766cba0dd44ec080420acc10990282",
  b: "ab766cba0dd44ec080420acc10990282"
});
console.log(results);

// =>
// {
//   "a": "{{{{ab766cba0dd44ec080420acc10990282.itemId}}.itemId}}",
//   "b": "{{{{ab766cba0dd44ec080420acc10990282.itemId}}.itemId}}"
// }
```

**Comment by rweber-esri** [comment]
@dbouwman @MikeTschudi 
I've updated this PR to point to `develop` and rebased, should be good to review.

**Pull request merged**